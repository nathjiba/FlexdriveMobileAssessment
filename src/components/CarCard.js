import React from 'react';
import {TouchableOpacity, Image, Text, View, Dimensions} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import sampleCar from '../assets/car_placeholder.png'
import lyfticon from '../assets/lyft_icon.png'
import PriceButton from '../components/PriceButton'

const {width, height} = Dimensions.get('window');

const CarCard = props => {
    return (
        <TouchableOpacity style={[styles.mainContainer]}>
            <View>
                <Image
                    source={{
                    uri: props.image.url
                }}
                    defaultSource={sampleCar}
                    style={styles.carImageStyle}></Image>

                {props.rideshareEligible
                    ? <View style={styles.eligibaleView}>
                            <Image
                                source={lyfticon}
                                style={{
                                height: 26,
                                width: 26,
                                resizeMode: 'contain'
                            }}></Image>
                            <Text style={styles.eligibleTextStyle}>ELIGIBEL</Text>
                        </View>
                    : <View/>}
            </View>
            <View style={{
                flexDirection: 'row'
            }}>

                <View style={styles.textParentStyle}>
                    <Text style={styles.carNameTextStyle} numberOfLines={2} ellipsizeMode={'tail'}>{`${props.year} ${props.make} ${props.model}`}</Text>
                    <Text
                        style={styles.carSubTitleTextStyle}
                        numberOfLines={1}
                        ellipsizeMode={'tail'}>{props.location.name}</Text>
                </View>
                <View style={styles.priceButtonParentStyle}>
                    <PriceButton price={props.price}></PriceButton>
                </View>
            </View>

        </TouchableOpacity>
    );
}

const styles = EStyleSheet.create({
    mainContainer: {
        flex: 1,
        shadowOffset: {
            height: 1
        },
        shadowColor: 'black',
        shadowOpacity: 0.2,
        backgroundColor: 'white',
        elevation: 2
    },
    headerTextStyle: {
        fontSize: 16,
        fontWeight: '600'
    },
    // TODO : I have used hardcoded image style, when I will get proper image like
    // 1x,2x and 3x, no need to give hardcoded size
    locationImageStyle: {
        height: 8,
        width: 8
    },
    buttonTextStyle: {
        fontSize: 13,
        fontWeight: '700',
        marginLeft: 5

    },
    buttonImageStyle: {
        height: 22,
        width: 22
    },
    carNameTextStyle: {
        fontSize: '1 rem',
        fontWeight: '600'
    },
    carSubTitleTextStyle: {
        fontSize: '0.7 rem',
        fontWeight: '600',
        color: '#818A8F'
    },
    textParentStyle: {
        flex: 0.65,
        marginLeft: 16,
        paddingRight: 16,
        justifyContent: 'center'
    },
    priceButtonParentStyle: {
        flex: width <= 320
            ? 0.45
            : 0.35,
        padding: 10
    },
    carImageStyle: {
        aspectRatio: 16 / 9,
        resizeMode: 'cover',
        height: undefined,
        width: undefined
    },
    eligibaleView: {
        backgroundColor: '$appPink',
        position: 'absolute',
        height: 32,
        borderTopRightRadius: 14,
        borderBottomRightRadius: 14,
        marginTop: height * 0.2,
        flexDirection: 'row',
        alignItems: 'center',
        paddingLeft: 10,
        paddingRight: 16
    },
    eligibleTextStyle: {
        marginLeft: 6,
        color: 'white',
        fontSize: 16,
        fontWeight: '600'
    }

});

export default CarCard;
