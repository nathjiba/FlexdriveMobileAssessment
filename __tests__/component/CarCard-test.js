import React from 'react';
import CarCard from '../../src/components/CarCard';
import renderer from 'react-test-renderer';
import TestData from '../../MockData'

test('should render without crash', () => {
    const {
        featureImage,
        location,
        rideshareEligible,
        model,
        pricing,
        year,
        make
    } = TestData.CAR_OBJECT
    const tree = renderer
        .create(<CarCard
            image={featureImage}
            location={location}
            rideshareEligible={rideshareEligible}
            model={model}
            price={pricing[0]}
            year={year}
            make={make}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});