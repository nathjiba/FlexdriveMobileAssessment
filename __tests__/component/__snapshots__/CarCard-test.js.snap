// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crash 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View>
    <Image
      defaultSource={
        Object {
          "testUri": "../../../src/assets/car_placeholder.png",
        }
      }
      source={
        Object {
          "uri": "https://f1-prod-assets.s3.amazonaws.com/f910b73b-f8a3-4e21-968e-acacd644ef25/images/IMG_4317/IMG_4317-2017-07-06-092338-full.jpg",
        }
      }
    />
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/lyft_icon.png",
          }
        }
        style={
          Object {
            "height": 26,
            "resizeMode": "contain",
            "width": 26,
          }
        }
      />
      <Text>
        ELIGIBEL
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View>
      <Text
        ellipsizeMode="tail"
        numberOfLines={2}
      >
        1990 Kia Cadenza
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
      >
        Flexdrive of Cherry Hill
      </Text>
    </View>
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text>
          $20,422
        </Text>
        <Text>
          For 7 days
        </Text>
      </View>
    </View>
  </View>
</View>
`;
